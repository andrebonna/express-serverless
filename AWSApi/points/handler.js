'use strict';

module.exports.create = (event, context, callback) => {
    const response = {
        statusCode: 200,
        body: JSON.stringify({
            message: event
        })
    };
    callback(null, response);
};
