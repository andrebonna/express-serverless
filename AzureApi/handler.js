'use strict';

/* eslint-disable no-param-reassign */

module.exports.hello = function (context) {
  context.log('JavaScript HTTP trigger function processed a request.');

  context.res = {
    // status: 200, /* Defaults to 200 */
    body: JSON.stringify(context),
  };

  context.done();
};
