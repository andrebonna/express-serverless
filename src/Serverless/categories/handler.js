import React from 'react';
import templateBuilder from '../commons/templateBuilder';
import constants from '../commons/constants';
import Categories from '../../View/categories/Categories';

export function get(event, context, callback) {
    templateBuilder({
        title: 'Categories', 
        metas: constants.metas,
        props: {
            children: <Categories />
        }
    }, (err, data) => {
        const response = {
            statusCode: 200,
            headers: {
                'Content-type': 'text/html'
            },
            body: data
        };
        callback(null, response);
    });
}
