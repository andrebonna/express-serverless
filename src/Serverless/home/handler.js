import React from 'react';
import AWS from 'aws-sdk';
import templateBuilder from '../commons/templateBuilder';
import constants from '../commons/constants';
import Home from '../../View/home/Home';

const dynamodb = new AWS.DynamoDB.DocumentClient({ region: 'us-east-1' });

function getImagesFiltered(queryStringParameters) {
    let params = {
        TableName: 'Photos',
        Select: 'ALL_ATTRIBUTES'
    };

    if (queryStringParameters && queryStringParameters.filter) {
        params = {
            ...params,
            ScanFilter: {
                Categories: {
                    ComparisonOperator: 'CONTAINS',
                    AttributeValueList: [queryStringParameters.filter]
                }
            }
        };
    }

    return dynamodb.scan(params).promise();
}

function getCategories(data) {
    return dynamodb.scan({
        TableName: 'Photos',
        Select: 'SPECIFIC_ATTRIBUTES',
        AttributesToGet: ['Categories']
    }).promise().then((categories)=>{
        const categoriesSet = new Set();
        categories.Items.map(({ Categories })=>{
            Categories.forEach(category => categoriesSet.add(category));
        });          
        return { 
            data, 
            categories: [...categoriesSet] 
        };
    });
}

export function get(event, context, callback) {

    const {
        queryStringParameters
    } = event;

    getImagesFiltered(queryStringParameters)
        .then(getCategories)
        .then(({data, categories}) => {
            templateBuilder({
                title: 'Andr√©a Buck Photos',
                metas: constants.metas,
                props: {
                    categories,
                    children: <Home images={data.Items} />
                }
            }, (err, data) => {
                if (err) {
                    callback(err);
                }
                const response = {
                    statusCode: 200,
                    headers: {
                        'Content-type': 'text/html'
                    },
                    body: data
                };
                callback(null, response);
            });
        }).catch(callback);
}
