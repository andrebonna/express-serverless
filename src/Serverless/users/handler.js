import React from 'react';
import ejs from 'ejs';
import ReactDOMServer from 'react-dom/server';
import Head from '../../View/Head';
import App from '../../View/App';

export function get(event, context, callback) {

    const head = React.createElement(Head, {
        title: 'Serverless Universal React',
        metas: event.queryStringParameters
    });
    const data = {
        cdn_path: process.env.CDN_PATH || '',
        head: ReactDOMServer.renderToString(head),
        content: ReactDOMServer.renderToString(React.createElement(App))
    };
    ejs.renderFile(`${__dirname}/../../View/template.ejs`, data, function(err, data) {
        const response = {
            statusCode: 200,
            headers: {
                'Content-type': 'text/html'
            },
            body: data
        };
        callback(null, response);
    });
}

export function create(event, context, callback) {
    const response = {
        statusCode: 200,
        body: event.body
    };
    callback(null, response);
}
