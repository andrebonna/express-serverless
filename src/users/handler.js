'use strict';

module.exports.create = (event, context, callback) => {
    const response = {
        statusCode: 200,
        body: event.body
    };
    callback(null, response);
};
