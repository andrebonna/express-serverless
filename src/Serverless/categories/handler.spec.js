import { get } from './handler';

describe('Handler Lambdas', ()=>{
    it('Test Get Categories Page', (done) => {
        get({}, null, (_, response)=>{
            expect(response.body).toMatchSnapshot();
            done();
        });
    });
});
