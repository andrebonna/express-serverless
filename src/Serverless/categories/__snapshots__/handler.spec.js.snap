exports[`Handler Lambdas Test Get Categories Page 1`] = `
"<!DOCTYPE html>
<html lang=\"en\">
    <head data-reactroot=\"\" data-reactid=\"1\" data-react-checksum=\"-1251837647\"><title data-reactid=\"2\">Categories</title><link rel=\"stylesheet\" type=\"text/css\" href=\"bundle.css\" data-reactid=\"3\"/><meta name=\"viewport\" content=\"width=device-width, initial-scale=1\" data-reactid=\"4\"/><meta name=\"keywords\" content=\"photos, nature, urban, photography, fotografia\" data-reactid=\"5\"/></head>
    <body>
        <div id=\"content\"><div data-reactroot=\"\" data-reactid=\"1\" data-react-checksum=\"-1924174018\"><nav class=\"navbar navbar-inverse navbar-fixed-top\" data-reactid=\"2\"><div class=\"container-fluid\" data-reactid=\"3\"><div class=\"navbar-header\" data-reactid=\"4\"><a href=\"#\" class=\"navbar-brand\" data-reactid=\"5\">React-Bootstrap</a><button type=\"button\" class=\"navbar-toggle collapsed\" data-reactid=\"6\"><span class=\"sr-only\" data-reactid=\"7\">Toggle navigation</span><span class=\"icon-bar\" data-reactid=\"8\"></span><span class=\"icon-bar\" data-reactid=\"9\"></span><span class=\"icon-bar\" data-reactid=\"10\"></span></button></div><div class=\"navbar-collapse collapse\" data-reactid=\"11\"><ul class=\"nav navbar-nav navbar-right\" data-reactid=\"12\"><li class=\"dropdown\" data-reactid=\"13\"><a id=\"basic-nav-dropdown\" role=\"button\" class=\"dropdown-toggle\" aria-haspopup=\"true\" aria-expanded=\"false\" href=\"#\" data-reactid=\"14\"><!-- react-text: 15 -->Categories<!-- /react-text --><!-- react-text: 16 --> <!-- /react-text --><span class=\"caret\" data-reactid=\"17\"></span></a><ul role=\"menu\" class=\"dropdown-menu\" aria-labelledby=\"basic-nav-dropdown\" data-reactid=\"18\"><li role=\"presentation\" class=\"\" data-reactid=\"19\"><a href=\"categories\" role=\"menuitem\" tabindex=\"-1\" data-reactid=\"20\">All</a></li><li role=\"presentation\" class=\"\" data-reactid=\"21\"><a href=\"categories?filter=urban\" role=\"menuitem\" tabindex=\"-1\" data-reactid=\"22\">Urban</a></li><li role=\"presentation\" class=\"\" data-reactid=\"23\"><a href=\"categories?filter=nature\" role=\"menuitem\" tabindex=\"-1\" data-reactid=\"24\">Nature</a></li></ul></li><li role=\"presentation\" class=\"\" data-reactid=\"25\"><a href=\"contact\" data-reactid=\"26\">Contact Us</a></li></ul></div></div></nav><section class=\"main-content\" data-reactid=\"27\"><div data-reactid=\"28\">Categories</div></section><footer class=\"footer\" data-reactid=\"29\"><p data-reactid=\"30\">© André Bonna</p></footer></div></div>
        <script type=\"text/javascript\">
            window.APP_PROPS = \'{\"children\":{\"key\":null,\"ref\":null,\"props\":{},\"_owner\":null,\"_store\":{}},\"categories\":[{\"name\":\"All\",\"href\":\"categories\"},{\"name\":\"Urban\",\"href\":\"categories?filter=urban\"},{\"name\":\"Nature\",\"href\":\"categories?filter=nature\"}],\"page\":\"Categories\"}\';
        </script>
        <script src=\"bundle.js\"></script>
    </body>
</html>"
`;
